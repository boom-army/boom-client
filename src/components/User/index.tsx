import { useContext } from "react";

import AccountBoxIcon from "@mui/icons-material/AccountBox";
import {
  Box,
  Card,
  CardActionArea,
  IconButton,
  Typography,
} from "@mui/material";
import { Follow } from "../Profile/Follow";
import { Link } from "react-router-dom";
import { ThemeContext } from "../../contexts/theme";
import { User as UserProps } from "../../generated/graphql";
import { UserAvatar } from "../UserAvatar";
import { Link as RouterLink } from "react-router-dom";

type Props = {
  user: UserProps;
};

export const User = ({ user }: Props) => {
  const { theme } = useContext(ThemeContext);
  return (
    <Card>
      <Box sx={{ display: "flex" }}>
        <CardActionArea
          component={RouterLink}
          to={`/${user.handle}`}
          sx={{ padding: "0.7em" }}
        >
          <Box sx={{ display: "flex", overflow: "hidden" }}>
            <Box>
              <UserAvatar
                sx={{
                  width: 30,
                  height: 30,
                }}
                avatar={user?.avatar}
                isNFT={user?.data?.avatarIsNFT ?? false}
              />
            </Box>
            <Box ml={2}>
              <Typography noWrap sx={{ width: "10em" }}>
                {user && user.consumerName}
              </Typography>
              <Typography
                noWrap
                sx={{ color: theme.secondaryColor, width: "10em" }}
              >{`@${user && user.handle}`}</Typography>
            </Box>
          </Box>
        </CardActionArea>
        <Box sx={{ minWidth: "40px" }}>
          <IconButton edge="end" aria-label="delete">
            {user && !user.isSelf ? (
              <Follow
                sm
                id={user && user.id}
                isFollowing={user && user.isFollowing}
              />
            ) : (
              <Link to="/settings/profile">
                <AccountBoxIcon />
              </Link>
            )}
          </IconButton>
        </Box>
      </Box>
    </Card>
  );
};
