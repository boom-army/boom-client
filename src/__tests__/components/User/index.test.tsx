import React from "react";
import { User } from "../../../components/User/index";
import renderer from "react-test-renderer";

// const data :any= {
//     "id": "ckzny1iv10031lmn568gj3b3n",
//     "avatar": "",
//     "handle": "shy-cloud-4965",
//     "consumerName": "shy-cloud-4965"
//     , "__typename": "User"
// }

const data: any = {
  id: "ckzny1iv10031lmn568gj3b3n",
  handle: "shy-cloud-4965",
  avatar: "",
  isFollowing: false,
  isSelf: true,
  consumerName: "shy-cloud-4965",
  bio: "full stack developer",
  __typename: "User",
};

describe("user info", () => {
  test("renders correctly user", () => {
    const tree = renderer.create(<User user={data} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
