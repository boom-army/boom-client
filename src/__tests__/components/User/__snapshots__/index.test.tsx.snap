// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user info renders correctly user 1`] = `
<div
  className="MuiGrid-root MuiGrid-container css-1cb2q4o-MuiGrid-root"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-10 css-17p3wh3-MuiGrid-root"
  >
    <div
      className="MuiBox-root css-k008qs"
    >
      <div
        className="MuiBox-root css-0"
      >
        <div
          className="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-1jl820a-MuiAvatar-root"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiAvatar-fallback css-10mi8st-MuiSvgIcon-root-MuiAvatar-fallback"
            data-testid="PersonIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
            />
          </svg>
        </div>
      </div>
      <div
        className="MuiBox-root css-d0uhtl"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1pp5b5c-MuiTypography-root"
        >
          shy-cloud-4965
        </p>
        <p
          className="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-1ny6etj-MuiTypography-root"
        >
          @shy-cloud-4965
        </p>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
  >
    <div
      className="MuiBox-root css-1a9getn"
    >
      <button
        aria-label="delete"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <a
          href="/settings/profile"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="AccountBoxIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 5v14c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2H5c-1.11 0-2 .9-2 2zm12 4c0 1.66-1.34 3-3 3s-3-1.34-3-3 1.34-3 3-3 3 1.34 3 3zm-9 8c0-2 4-3.1 6-3.1s6 1.1 6 3.1v1H6v-1z"
            />
          </svg>
        </a>
      </button>
    </div>
  </div>
</div>
`;
